src/processgraph.js

```diff
@@ -473,21 +478,23 @@ class ProcessGraph {
 	 * Gets the process for the given process ID or node.
 	 * 
 	 * @param {ProcessGraphNode|string} id 
+	 * @param {?string} [namespace=null]
 	 * @returns {object|null}
 	 * @throws {ProcessGraphError} - ProcessUnsupported
 	 */
-	getProcess(id) {
+	getProcess(id, namespace = null) {
 		if (this.processRegistry === null) {
 			return null;
 		}
 		if (id instanceof ProcessGraphNode) {
 			id = id.process_id;
+			namespace = id.namespace;
```

Comment:
How can a variable named `id` have attributes? That's not very intuitive.

Reply:
Indeed, made this more intuitive.